### PR TITLE
Remove coveralls, add codecov

### PR DESCRIPTION
**Pull request by kaustubh-nair**
Partially fixes https://github.com/publiclab/plots2/issues/5931

### CODE REVIEW

**Comment by kaustubh-nair** [comment]
@jywarren @publiclab/reviewers What are these lines supposed to be doing? I'm not sure if they are related to coveralls or not
https://github.com/publiclab/plots2/blob/f747a25e10f92ed2062772121fda759d5ccb7b30/.travis.yml#L40-L44

File: Rakefile [hunk 1]
```diff
@@ -20,8 +20,6 @@ namespace :test do
 
   desc "Run rails and teaspoon tests"
   task :all => :environment do
-    require 'coveralls/rake/task'
-    Coveralls::RakeTask.new
     if ENV['GENERATE_REPORT'] == 'true'
       require 'ci/reporter/rake/test_unit'
       Rake::Task["ci:setup:testunit"].execute
```
Comment (kaustubh-nair) [inline Rakefile]: @jywarren need some info on these tasks

File: .travis.yml [hunk 2]
```diff
@@ -40,11 +38,11 @@ install:
...
-  - docker-compose exec web bash -c "CI=TRUE GENERATE_REPORT=true $TASK"
-  - if [ $TASK != "rake test:all" ]; then
-      echo -e '<?xml version="1.0" encoding="UTF-8"?>' > output.xml;
-      tail -n +2 -q ./test/reports/TEST*.xml >> output.xml;
-    fi
+  - docker-compose exec web bash -c "GENERATE_REPORT=true $TASK"
+# - if [ $TASK != "rake test:all" ]; then
+#     echo -e '<?xml version="1.0" encoding="UTF-8"?>' > output.xml;
+#     tail -n +2 -q ./test/reports/TEST*.xml >> output.xml;
+#   fi
```
Comment (kaustubh-nair) [inline .travis.yml]: @jywarren @publiclab/reviewers What are these lines supposed to be doing? I'm not sure if they are related to coveralls or not

**Comment by kaustubh-nair** [comment]
I've got codecov setup but the build is failing, because of some errors in the Rakefile I think
@jywarren Can you have a look?

File: .travis.yml (hunk 2 above)
Comment (alaxalves) [reply .travis.yml]: They're related to Coveralls, they generate the XML to be pushed to Coveralls. You can remove those.

Comment (kaustubh-nair) [reply .travis.yml]: Not sure about this. Dangerbot uses the xml too I think.

Comment (jywarren) [reply .travis.yml]: These were intended to search through standard formatted JUnit test reports and copy them into the Dangerbot comment in the PR. They broke some time ago, but were a really nice way to help people see where errors were occurring. I'd like to leave them in so we can fix them again some day!

**Comment by jywarren** [comment]
Ah, i think you'll have to remove `Coveralls.push` - 

```

rake aborted!
Don't know how to build task 'coveralls:push' (See the list of available tasks with `rake --tasks`)
/app/Rakefile:29:in `block (2 levels) in <top (required)>'
/usr/local/bundle/gems/rake-12.3.2/exe/rake:27:in `<top (required)>'
Tasks: TOP => test:all
(See full trace by running task with --trace)
The command "docker-compose exec web bash -c "GENERATE_REPORT=true $TASK"" exited with 1.
```

Probably in the rakefile, line 29?

**Comment by kaustubh-nair** [comment]
Not sure why project check is turning up even though it's off. I guess it should be fine after merging?
@jywarren This is ready
Thanks!

**Comment by kaustubh-nair** [comment]
@jywarren On second thought, let's not merge this till I fix the inconsistencies on mapknitter. I might have to change the reporter

**Comment by gauravano** [comment]
For sure, the error is not related to your code. I have restarted the job.

**Comment by kaustubh-nair** [comment]
We're facing the same coverage drop here too :/
I'm starting to think this a bug with Codecov itself and not an issue on our side

**Comment by jywarren** [comment]
Is the coverage drop consistent? I.e. if it's not related to code, are we seeing a non-deterministic behavior, perhaps resulting from the order of completion of tests, or from other branch tests getting mixed up somehow? Or is it showing the same exact drop each time this is run?

Can you open an issue on CodeCov's tracker and link between here and there to try to get some help on this? Thank you!

**Comment by jywarren** [comment]
Let's see how this does and then we can resolve conflicts and merge!

**Comment by jywarren** [comment]
@Uzay-G can you see why the tests didn't pass here? https://travis-ci.org/publiclab/plots2/builds/630645697?utm_source=github_status&utm_medium=notification

I restarted it after resolving conflicts but not sure why it had failed before. I hope it doesn't this time!

**Comment by Uzay-G** [comment]
I looked at it but don't understand why it bugged out then and worked now although nothing was changed. The error just says `Mysql2::Error::ConnectionError: Can't connect to MySQL server on '127.0.0.1' (111 "Connection refused")`

**Comment by jywarren** [comment]
Wow! This would increase our coverage to 80% then... `Merging #5954 into master will increase coverage by 6.08%.`

This would be awesome. Do you think it's ready to merge then? 

From the chatroom: 

> @Uzay-G will try to take a look. it could just need to be restarted by closing and reopening... could be a random timeout

**Comment by jywarren** [comment]
Let's do this!!!

**Comment by jywarren** [comment]
🎉 🎉 🎉 🎉 🎉

**Comment by Uzay-G** [comment]
Nice! Yeah I think it's ready and it should work! :balloon: :+1:

**Comment by Uzay-G** [comment]
Ah! I don't understand why but for some reason the codecov bot is not commenting on new pull requests. This issue is really a mystery...

**Comment by Uzay-G** [comment]
Ok actually it worked but there is a problem. See here for what it did: https://github.com/publiclab/plots2/pull/7086. It returns the error: exclamation No coverage uploaded for pull request base (master@e16e4bc).
I will try to fix it

**Comment by Uzay-G** [comment]
This is the error:
codecov
 
I think we need to edit the settings on the codecov dashboard so that the base is more recent maybe?? 
I'm not sure because I don't have permission to see the settings

**Comment by jywarren** [comment]
Ah, it's supposed to compare to the base that is the HEAD position of the
master branch, not a static version, right?

**Comment by Uzay-G** [comment]
Yeah it needs to compare to the base. What is it comparing to right now?

**Comment by jywarren** [comment]
I don't know, but is it potentially an issue that if tests haven't changed
at all, then the PR doesn't meet the minimum requirement?

**Comment by Uzay-G** [comment]
I don't think it's because of the pull request changes because all of the pull requests have the same message. The problem is that it is comparing to a commit that doesn't have a codecov report so it can't compare the changes.

**Comment by Uzay-G** [comment]
This page might help us: https://docs.codecov.io/docs/comparing-commits

**Comment by Uzay-G** [comment]
I think I know the problem and will try to fix it when I come back

**Comment by jywarren** [comment]
OK, awesome!

**Comment by jywarren** [comment]
Hi @Uzay-G (ref #5954) this is a good example PR: https://github.com/publiclab/plots2/pull/6957

> ❗️ No coverage uploaded for pull request base (master@b701668). Click here to learn what that means.

image

Does this mean that as soon as we merge another PR to master, it'll have a recorded coverage base for that commit?

OH! You know what? What if squashing and merging is causing a new commit hash to be generated, and so we have to be sure tests are run on that too so that it files a coverage report for the newly generated squashed commit???

**Comment by Uzay-G** [comment]
Yeah maybe we need to merge one to have a base with a coverage report. I'm not sure what you mean by the second part bur merging one might be a good idea.

**Comment by Uzay-G** [comment]
We could merge #7076 to try it out

**Comment by jywarren** [comment]
OK i merged https://github.com/publiclab/plots2/pull/7089

Let's look again. If it doesn't work, we can try merging WITHOUT squashing to test if that does not generate a different commit and so the coverage would be saved???

**Comment by Uzay-G** [comment]
It worked here: https://github.com/publiclab/plots2/pull/5724#issuecomment-570382680. I think the issue is resolved!

**Comment by Uzay-G** [comment]
Since you merged the pr, codecov is working. Yay we fixed it! :birthday:

**Comment by jywarren** [comment]
Ah! I think one complicating factor is that we are seeing it update
progressively as the tests run...! We may just need to wait a bit longer.

**Comment by Uzay-G** [comment]
Yeah I agree

**Pull request merged**